shanoir-ng-datasets/src/main/java/org/shanoir/ng/importer/service/ImporterService.java

```diff
@@ -249,26 +298,37 @@ private Set<DatasetAcquisition> generateAcquisitions(Examination examination, Im
     }
 
     private QualityCardResult checkQuality(Examination examination, ImportJob importJob) throws ShanoirException {
-        Attributes dicomAttributes = null;
-        try {
-            Serie firstSerie = importJob.getFirstSerie();
-            if (firstSerie == null) {
-                throw new ShanoirException("The given import job does not provide any serie. Examination : " + examination.getId());
-            }
-            dicomAttributes = dicomProcessing.getDicomObjectAttributes(firstSerie.getFirstDatasetFileForCurrentSerie(), firstSerie.getIsEnhanced());
-        } catch (IOException e) {
-            throw new ShanoirException("Unable to retrieve dicom attributes for examination " + examination.getId(), e);
+        ExaminationAttributes<String> dicomAttributes = null;          
+        Study firstStudy = importJob.getFirstStudy();
+        if (firstStudy == null) {
+            throw new ShanoirException("The given import job does not provide any serie. Examination : " + examination.getId());
         }
+        dicomAttributes = dicomProcessing.getDicomExaminationAttributes(firstStudy);
         List<QualityCard> qualityCards = qualityCardService.findByStudy(examination.getStudyId());
         QualityCardResult qualityResult = new QualityCardResult();
         for (QualityCard qualityCard : qualityCards) {
             if (qualityCard.isToCheckAtImport()) {
-                qualityResult.merge(qualityCard.apply(examination, dicomAttributes));                       
+                qualityResult.merge(qualityCard.apply(examination, dicomAttributes, downloader));                       
             }
         }
         return qualityResult;
     }
 
+    private QualityCardResult checkQuality(Examination examination, Set<DatasetAcquisition> limitToTheseAcquisitions, ImportJob importJob) throws ShanoirException {
+        // save the exam acquisitions
+        List<DatasetAcquisition> saveList = new ArrayList<>();
```

Comment:
Hi Julien, thank you for your corrections. I am wondering about the below code. The ImporterService is only responsible for new-imported-data (either on a new and empty exam or an existing exam). The below code is really quite a work-around (sorry for the word), but putting existing acquisitions into memory, simulate a new exam and then putting back the old acquisitions, is really strange. I have fear in two months we will not understand the code ourselves. Please let's discuss this and how we continue from here. Thank you and with kind regards, Michael